disnake/ext/mypy_plugin/__init__.py

```diff
@@ -1,68 +1,28 @@
 # SPDX-License-Identifier: MIT
 
 import typing as t
+import warnings
 
-from mypy.plugin import AnalyzeTypeContext, Plugin
-from mypy.types import AnyType, EllipsisType, RawExpressionType, Type, TypeOfAny
+from mypy.plugin import Plugin
 
 
 class DisnakePlugin(Plugin):
-    def get_type_analyze_hook(
-        self, fullname: str
-    ) -> t.Optional[t.Callable[[AnalyzeTypeContext], Type]]:
-        if fullname == "disnake.ext.commands.params.Range":
-            return range_type_analyze_callback
-        if fullname == "disnake.ext.commands.params.String":
-            return string_type_analyze_callback
-        return None
-
-
-def range_type_analyze_callback(ctx: AnalyzeTypeContext) -> Type:
-    args = ctx.type.args
-
-    if len(args) != 2:
-        ctx.api.fail(f'"Range" expected 2 parameters, got {len(args)}', ctx.context)
-        return AnyType(TypeOfAny.from_error)
-
-    for arg in args:
-        if isinstance(arg, EllipsisType):
-            continue
-        if not isinstance(arg, RawExpressionType):
-            ctx.api.fail('invalid usage of "Range"', ctx.context)
-            return AnyType(TypeOfAny.from_error)
-
-        name = arg.simple_name()
-        # if one is a float, `Range.underlying_type` returns `float`
-        if name == "float":
-            return ctx.api.named_type("builtins.float", [])
-        # otherwise it should be an int; fail if it isn't
-        elif name != "int":
-            ctx.api.fail(f'"Range" parameters must be int or float, not {name}', ctx.context)
-            return AnyType(TypeOfAny.from_error)
-
-    return ctx.api.named_type("builtins.int", [])
-
-
-def string_type_analyze_callback(ctx: AnalyzeTypeContext) -> Type:
-    args = ctx.type.args
-
-    if len(args) != 2:
-        ctx.api.fail(f'"String" expected 2 parameters, got {len(args)}', ctx.context)
-        return AnyType(TypeOfAny.from_error)
-
-    for arg in args:
-        if isinstance(arg, EllipsisType):
-            continue
-        if not isinstance(arg, RawExpressionType):
-            ctx.api.fail('invalid usage of "String"', ctx.context)
-            return AnyType(TypeOfAny.from_error)
-
-        name = arg.simple_name()
-        if name != "int":
-            ctx.api.fail(f'"String" parameters must be int, not {name}', ctx.context)
-            return AnyType(TypeOfAny.from_error)
-
-    return ctx.api.named_type("builtins.str", [])
+    """Deprecated mypy plugin."""
```

Comment:
Honestly, lets not deprecate this, in case we need to add a mypy plugin again in the future. Users who use mypy have already started to use this, so lets keep it existing. That doesn't mean it has to do anything, but we don't need to deprecate it either.

Reply:
I didn't want to completely remove it for that exact reason, but right now there's no reason for users to keep it enabled.
If we do need a mypy plugin again in the future, the typing system is probably being misused in some way :p  Notably, plugins for pyright aren't a thing, so this would really only apply in cases where mypy is misbehaving, in which case it's not our job to fix it, imo.

Reply:
Right, but it would still be a good idea to not document it as deprecated (yet) but rather a no-op plugin for the time being.

Reply:
alright. https://github.com/DisnakeDev/disnake/pull/991/commits/a3e00ece5fa1f1231ab219f5eb233c6a46cb4000